💄 Rework the bookmark widget

Tagaishi opened the pull request:
### Category
> Cosmetic

### Overview
> Redesigned the bookmark widget:
> - Take less space per item.
> - Remove the bubble around each items in the vertical and horizontal option to prefer a simple line separator
> - Have the option to hide the Hostname.
> - Have the option to hide the Icon.
> - Completely fill the space available in the widget. (Thanks to @manuel-rw for battling mantine with me on that one)
> - Highlight currently hovered item using user primary shade (at 25% opacity to prevent overdoing it).

### Issue Number
> Closes #1035 

### Screenshot
> Grid using all the space available:
![image](https://github.com/ajnart/homarr/assets/26098587/9b3b4df8-cb3d-4ee7-82ad-d6ddba84573e)

> Vertical and horizontal underflow stretch:
![image](https://github.com/ajnart/homarr/assets/26098587/686b8f0d-6989-40d9-b871-7836e76a64a0)
![image](https://github.com/ajnart/homarr/assets/26098587/e6edd556-30d6-4c09-8c32-13b58068cd45)

> Vertical and horizontal overflows:
![image](https://github.com/ajnart/homarr/assets/26098587/16371f20-c1eb-4613-9539-37e98508e59f)
![image](https://github.com/ajnart/homarr/assets/26098587/d3833eeb-9737-4e3e-bb5a-1934acd27cbc)

> All images above also show the highlight function, light/dark theme variation and all links hidden except for radarr.


> Options Hide link
![image](https://github.com/ajnart/homarr/assets/26098587/a8a1f7b9-2d42-430d-9d9e-3e2d81d9f37b)

Tagaishi commented:
Grid items weren't centered height-wise so I corrected it quickly.
![image](https://github.com/ajnart/homarr/assets/26098587/5fe32fb1-6878-4dd2-bdbd-a6006e4a01d9)

manuel-rw commented on src/widgets/bookmark/BookmarkWidgetTile.tsx:
```diff
@@ -39,6 +40,7 @@ interface BookmarkItem {
   href: string;
   iconUrl: string;
   openNewTab: boolean;
+  hideLink: boolean;
```
Rename to ``hideHostname`` or ``hideBasename``. Link is the whole sequence - we only display a part of it.

manuel-rw commented on src/widgets/bookmark/BookmarkWidgetTile.tsx:
```diff
@@ -226,26 +241,44 @@ function BookmarkWidgetTile({ widget }: BookmarkWidgetTileProps) {
+          styles={{
+            viewport:{
+              //mantine being mantine again... this might break
+              '& div[style="min-width: 100%; display: table;"]':{
```
@Meierschlumpf I wrote this together yesterday with OP. This is the only way, right?

Meierschlumpf replied:
I guess it's fine

ajnart commented:
My only review is that the rounded corners should be “md” on the cards, otherwise it looks good to me :)

Tagaishi commented:
Added a title option for people wishing to organize multiple different group of bookmarks and reworked the divider to be an independent element instead of the border of the bookmark itself.
Title is optional, if left empty it won't show like the original bookmark widget.
![image](https://github.com/ajnart/homarr/assets/26098587/ec636712-2b83-492d-b004-3488ccfd6cba)

Tagaishi commented:
Hide Icon like so:
![image](https://github.com/ajnart/homarr/assets/26098587/feb54ddc-c6e4-4a8e-b890-74fd22c075df)
![image](https://github.com/ajnart/homarr/assets/26098587/0af0f89d-0b46-499a-bbad-485d228560b0)

as requested in the issue 1035

ajnart commented:
@Tagaishi you can put the # to mention the issue directly or write 
"Closes #1035" and it will close it when this PR is merged

manuel-rw commented on src/widgets/bookmark/BookmarkWidgetTile.tsx:
```diff
@@ -206,72 +229,120 @@ function BookmarkWidgetTile({ widget }: BookmarkWidgetTileProps) {
+  const { colorScheme } = useMantineTheme();
+  return (
+  <Group spacing="0rem 1rem">
+    <Image hidden={item.hideIcon} src={item.iconUrl} width={47} height={47} fit="contain" withPlaceholder />
```
Those are too many attributes on one line. Use the Prettier extension from the VSC marketplace to format automatically.
You can see here, what extensions we mostly use:
https://homarr.dev/docs/community/developer-guide#development-environments--extensions
